### ISSUE TITLE
html language injection inside macros 

### ISSUE THREAD

**Issue by ta32** [intellij-rust/intellij-rust#6393]
![image](https://user-images.githubusercontent.com/26617779/99147420-eb727080-26d4-11eb-8a1f-9ec05e5efdd3.png)

would it be possible to implement language injection that detects html inside rust macros. This would be very useful. 

![image](https://user-images.githubusercontent.com/26617779/99147444-14930100-26d5-11eb-93a7-cb9bc8805231.png)

it looks like we can only define rules for js python, sql and xml files
and literal strings in other projects?

**Comment by Undin** [comment]
Technically, it's possible to support `Language Injection` rules for Rust but it requires some work.
But it seems language injection inside macro call body (not in a single string literal) is a harder task and it requires some investigation.
Also, currently, we don't tend to support Rust frameworks since they are quite unstable and maintenance can require a lot of work.
So I'm not sure that it will be implemented in the nearest future

**Comment by Undin** [comment]
Also, there is an idea to implement API for Rust libraries to provide info for IDEs about semantic of macro, for example, language

https://www.reddit.com/r/rust/comments/jv7sg3/intellij_rust_changelog_135/gcj3v2r/?context=3

**Comment by vlad20012** [comment]
> Technically, it's possible to support Language Injection rules for Rust but it requires some work.

It's relatively simple, but I'd say, the problem is that it's not HTML, but a mix of HTML and Rust. Naive HTML injection will result to lots of syntax errors.

**Comment by ricardoboss** [comment]
Any updates on this?

**Comment by vlad20012** [comment]
Nope. `yew` is not a pure `html`, it's a mix of html and Rust code, so it would not be very useful to blindly injecting it.

**Comment by subash-ev** [comment]
any updates ??
